crates/services/gas_price_service/src/v0/service.rs

```diff
@@ -194,17 +167,48 @@ mod tests {
         v0::{
             metadata::V0Metadata,
             service::GasPriceServiceV0,
-            uninitialized_task::initialize_algorithm,
+            uninitialized_task::{
+                initialize_algorithm,
+                SharedV0Algorithm,
+            },
         },
     };
     use fuel_core_services::{
+        RunnableService,
         Service,
         ServiceRunner,
+        StateWatcher,
     };
     use fuel_core_types::fuel_types::BlockHeight;
     use std::sync::Arc;
     use tokio::sync::mpsc;
 
+    #[async_trait::async_trait]
+    impl<L2, Metadata> RunnableService for GasPriceServiceV0<L2, Metadata>
```

Comment:
Can we run the task in the tests without the `ServiceRunner`?

Reply:
https://github.com/FuelLabs/fuel-core/blob/6979ce6f45a3813ca698f57607db932c8b43f222/crates/services/src/service.rs#L298 needs a `ServiceRunner` to `run_task`, which is private.

Reply:
Ah. But wait. Can't we just test `UninitializedTask`? That's the true interface that we deal with, not the task anyway.

Reply:
yes, but we want to test the behaviour of the task itself here

Reply:
there's a pending PR which will have merge conflicts if I add tests for `UninitializedTask` in this PR. let's treat it as a follow up.

Reply:
That works for me. 

> yes, but we want to test the behaviour of the task itself here

Just if there is no way to run a task without a service, then it's not a public interface and we should treat the service as the interface for that task.

Reply:
agreed